docs/opentelemetry.md

```diff
@@ -0,0 +1,68 @@
+# OpenTelemetry
+
+[OpenTelemetry](https://opentelemetry.io/) is a set of libraries, agents, and instrumentation to provide observability (metrics, logs and traces) in applications.
+
+`containerd-shim-wasm` crate has a set of APIs to enable OpenTelemetry tracing in the shim. This document is a guide on how to use OpenTelemetry tracing in your shim.
+
+## Usage
+
+To use OpenTelemetry tracing in your shim, you need to use the `opentelemetry` feature in the `containerd-shim-wasm` crate and the `tokio` crate.
+```toml
+containerd-shim-wasm = { workspace = true, features = ["opentelemetry"] }
+tokio = { version = "1", features = ["full"] }
+```
+
+Then, you may use the `containerd_shim_wasm::sandbox::cli::shim_main_with_otel` function to run the shim with OpenTelemetry tracing.
+
+```rust
+#[tokio::main]
+async fn main() {
+    shim_main_with_otel::<WasmtimeInstance>("wasmtime", version!(), revision!(), "v1", None);
+}
+```
+
+You may also use the `containerd_shim_wasm::sandbox::shim::OtelConfig` struct to configure the OpenTelemetry tracing manually.
+
+### Running containerd with OpenTelemetry
+
+You can run containerd with OpenTelemetry tracing by setting the following environment variables.
+
+```sh
+OTEL_EXPORTER_OTLP_ENDPOINT=http://localhost:4317
+OTEL_EXPORTER_OTLP_TRACES_PROTOCOL=grpc
```

Comment:
Why GRPC?
otlp default is http and I think jaeger expects http.

Reply:
jeager can work on both. but your point is valid, I will add grpc env var as well

Reply:
Just added `OTEL_EXPORTER_OTLP_PROTOCOL` to configure HTTP vs. gRPC